aws/sdk/integration-tests/s3/tests/express.rs

```diff
@@ -37,3 +37,64 @@ async fn list_objects_v2() {
         .await
         .unwrap();
 }
+
+#[tokio::test]
+async fn list_objects_v2_in_both_express_and_regular_buckets() {
+    let _logs = capture_test_logs();
+
+    let http_client = ReplayingClient::from_file(
+        "tests/data/express/list_objects_v2_in_both_express_and_regular_buckets.json",
+    )
+    .unwrap();
+    let config = aws_config::from_env()
+        .http_client(http_client.clone())
+        .no_credentials()
+        .region("us-west-2")
+        .load()
+        .await;
+    let config = Config::from(&config)
+        .to_builder()
+        .with_test_defaults()
+        .build();
+    let client = aws_sdk_s3::Client::from_conf(config);
+
+    // A call to an S3 Express bucket where we should see two request/response pairs,
+    // one for the `create_session` API and the other for `list_objects_v2` in S3 Express bucket.
+    let result = client
+        .list_objects_v2()
+        .bucket("s3express-test-bucket--usw2-az1--x-s3")
+        .send()
+        .await;
+    dbg!(result).expect("success");
+
+    // A call to a regular bucket, and request headers should not contain `x-amz-s3session-token`.
+    let result = client
+        .list_objects_v2()
+        .bucket("regular-test-bucket")
+        .send()
+        .await;
+    dbg!(result).expect("success");
+
+    // A call to another S3 Express bucket where we should again see two request/response pairs,
+    // one for the `create_session` API and the other for `list_objects_v2` in S3 Express bucket.
+    let result = client
+        .list_objects_v2()
+        .bucket("s3express-test-bucket-2--usw2-az3--x-s3")
+        .send()
+        .await;
+    dbg!(result).expect("success");
+
+    // This call should be an identity cache hit for the first S3 Express bucket,
+    // thus no HTTP request should be sent to the `create_session` API.
+    let result = client
+        .list_objects_v2()
+        .bucket("s3express-test-bucket--usw2-az1--x-s3")
+        .send()
+        .await;
+    dbg!(result).expect("success");
+
+    http_client
+        .validate_body_and_headers(Some(&["x-amz-s3session-token"]), "application/xml")
```

Comment:
if you want to validate this header, you can use a custom timesource to stop time. Then I think it will be deterministic?

Reply:
I believe a header `x-amz-s3session-token` is included for `validate_body_and_headers`, maybe you're referring to [assert_requests_match](https://github.com/smithy-lang/smithy-rs/blob/main/rust-runtime/aws-smithy-runtime/src/client/http/test_util/replay.rs#L220)?